tests/lxd/test_lxd_backend.cpp

```diff
@@ -531,32 +602,36 @@ TEST_F(LXDBackend, lxd_request_wrong_json_throws)
         std::runtime_error);
 }
 
-TEST_F(LXDBackend, suspend_while_stopped_keeps_same_state)
+TEST_F(LXDBackend, unsupported_suspend_throws)
 {
     mpt::StubVMStatusMonitor stub_monitor;
 
-    ON_CALL(*mock_network_access_manager.get(), createRequest(_, _, _)).WillByDefault([](auto, auto request, auto) {
-        auto op = request.attribute(QNetworkRequest::CustomVerbAttribute).toString();
-        auto url = request.url().toString();
+    ON_CALL(*mock_network_access_manager.get(), createRequest(_, _, _))
+        .WillByDefault([](auto, auto request, auto outgoingData) {
+            outgoingData->open(QIODevice::ReadOnly);
+            auto data = outgoingData->readAll();
+            auto op = request.attribute(QNetworkRequest::CustomVerbAttribute).toString();
+            auto url = request.url().toString();
 
-        if (op == "GET" && url.contains("1.0/virtual-machines/pied-piper-valley/state"))
-        {
-            return new mpt::MockLocalSocketReply(mpt::vm_state_stopped_data);
-        }
+            if (op == "GET" && url.contains("1.0/virtual-machines/pied-piper-valley/state"))
+            {
+                return new mpt::MockLocalSocketReply(mpt::vm_state_fully_running_data);
+            }
+            else if (op == "PUT" && url.contains("1.0/virtual-machines/pied-piper-valley/state") &&
+                     data.contains("stop"))
+            {
+                return new mpt::MockLocalSocketReply(mpt::stop_vm_data);
+            }
 
-        return new mpt::MockLocalSocketReply(mpt::not_found_data, QNetworkReply::ContentNotFoundError);
-    });
+            return new mpt::MockLocalSocketReply(mpt::not_found_data, QNetworkReply::ContentNotFoundError);
+        });
 
     mp::LXDVirtualMachine machine{default_description, stub_monitor, mock_network_access_manager.get(), base_url};
 
-    ASSERT_EQ(machine.current_state(), mp::VirtualMachine::State::stopped);
-
-    machine.suspend();
-
-    EXPECT_EQ(machine.current_state(), mp::VirtualMachine::State::stopped);
+    EXPECT_THROW(machine.suspend(), std::runtime_error);
```

Comment:
Don't want to be a bit more selective? Check the message?

Reply:
You can't (easily) check the message when using `EXPECT_THROW`.  You can only check the type.  The only way to more selective to use a different exception type and expect that.

Reply:
Yeah, which I commented a bit more about in #1604 review. I feel like this isn't dissimilar to catching a `std::runtime_error` - a bit broad…

We could use the mock logger to check the message, though, couldn't we?

Reply:
Actually, as @ricab mentioned in chat, there are the exception + message checks in `extra_assertions.h`.  I will use that.

Reply:
Ok, I think keeping it a `std::runtime_error` is fine here, but the test now checks the string to make sure it's the expected message since that is really what we care about.